lib/ansible/vars/manager.py

```diff
@@ -388,18 +383,15 @@ def plugins_by_groups():
                 raise AnsibleParserError("Error while reading vars files - please supply a list of file names. "
                                          "Got '%s' of type %s" % (vars_files, type(vars_files)))
 
-            # We now merge in all exported vars from all roles in the play,
-            # unless the user has disabled this
-            # role is public and
-            #    either static or dynamic and completed
-            # role is not set
-            #    use config option as default
-            for role in play.get_roles():
-                role_is_static_or_completed = role.static or role._completed.get(host.name, False)
-                if role.public and role_is_static_or_completed or \
-                   role.public is None and not C.DEFAULT_PRIVATE_ROLE_VARS and role_is_static_or_completed:
-
-                    all_vars = _combine_and_track(all_vars, role.get_vars(include_params=False, only_exports=True), "role '%s' exported vars" % role.name)
+            # We now merge in all exported vars from public roles in the play
+            if host:
+                for role in play.get_roles():
+                    if role.static or role._completed.get(host.name, False):
+                        all_vars = _combine_and_track(
+                            all_vars,
+                            role.get_vars(include_params=False, only_exports=True),
+                            f"role '{role.name}' exported vars"
+                        )
```

Comment:
Could you dedent this to make the patch smaller and less nested?
```suggestion
            for role in play.get_roles() if host or ():
                if role.static or role._completed.get(host.name, False):
                    all_vars = _combine_and_track(
                        all_vars,
                        role.get_vars(include_params=False, only_exports=True),
                        f"role '{role.name}' exported vars"
                    )
```

Reply:
no, as I comment above, the `if host` is incorrect